### DISCUSSION TITLE
Retrieval of data from an orientation map under the presence of drifting

### DISCUSSION THREAD

**Discussion by mfuhr-mpg** [mtex-toolbox/mtex#1135] [Ask Anything]
Hi,

I am currently trying to recover the reasonable orientation information from an orientation map whose quality suffered from drifting issues during the acquisition. The reason for the drifting probably was some thermal instability or an insufficient electrical contact. Prior to redoing the measurements, I would like to try and recover the information that is not affected by the drifting.
It is quite obvious to the eye that drifting can be identified by a repetition of a scanned line. This is why my approach would be to compare the lines and if more than for example 95 % of the orientations measured along the line are the same as another second line, I would delete all data between the initial line used for comparison and the second line (including this second line). After deleting, one would naturally stitch the orientation map so that there are no holes. I have already tried a few things, but it doesn't work out perfectly so far. One thing I struggled with was to choose a proper orientation definition (Euler angles, quaternions, etc.) in order to unambiguously compare the orientation measured along one line. Do you have any suggestions? Furthermore, I would be grateful for any other ideas in order to get rid of the drifting artefacts. You can download an example measurement from https://datashare.mpcdf.mpg.de/s/ZBUBKWHtvaaqO22.

Thanks in advance for your help!

Best regards,
Max Fuhr

**Comment by kilir** [comment]
Hi Max,
one way to compare orientations would be
`angle( list_of_ori1 , list_of_ori2  ) < some_threshold`, take list_of_ori1/2 form the ebsd with a common y-coordinate and once 5% are false, remember the y-coordinate for deletion and just assign new y-coordinates after having gone through all lines.

There's also `==` for orientations, however this would assume that the solutions are absolutely identical which is probably not the case. `eq(ori1,ori2,'tolerance',xy)` is another way.

However, I'm wondering, once you are drifting whether it makes sense at all to trust the coordinates as long as you do not have any reference shape.

Cheers,
Rüdiger

**Reply by mfuhr-mpg** [reply]
Hi Rüdiger,

thanks for the fast and on point answer. I will definitely try the workarounds you mentioned.
Your comment on whether I can trust the data is also very sound. If I think about it, it would probably be the best to only take the data above the first drifting artefact (assuming that the map was acquired from "top" to "bottom") and discard the rest. What do you think?

Best,
Max

**Reply by kilir** [reply]
Hi Max,
I agree, however if the  shifts are minor and depending on the question you'd like to answer, the impact might not be huge. If you'd like to say something about grain shapes or similar, discarding everything below sounds like a good idea.
Cheers,
Rüdiger

**Reply by mfuhr-mpg** [reply]
Hi Rüdiger,
great, I will try to treat the data with utmost care ;)
Thanks again!
Max
